pkg/v1/content/content.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2018 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package content allows you to create images directly from contents.
+package content
+
+import (
+	"archive/tar"
+	"bytes"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/empty"
+	"github.com/google/go-containerregistry/pkg/v1/mutate"
+	"github.com/google/go-containerregistry/pkg/v1/tarball"
+)
+
+// Image creates a image with the given contents. These images are reproducible and should be consistent.
```

Comment:
Is this true? Iterating over a map like we do on line 35 isn't guaranteed to be in the same order, so I think we could end up with different images from the same content.

Reply:
We might want to make `mutate.Canonical` sort the files in each layer.

Reply:
Though I'd err towards *also* doing [this](https://stackoverflow.com/a/18342865/6220920) for whatever code implements the
```go
func(map[string][]byte) v1.Layer
```
part.

Reply:
Yeah - this needs to be sorted. I thought this would be caught by the tests, but it wasn't. I added a new test which appears to fail :D.

Reply:
Done. Sorting is added and some more tests are added.